src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2658,6 +2658,310 @@ def update_nsg_rule_2017_03_01(instance, protocol=None, source_address_prefix=No
+    from azure.mgmt.privatedns import PrivateDnsManagementClient
+    from azure.mgmt.privatedns.models import PrivateZone
+    client = get_mgmt_service_client(cmd.cli_ctx, PrivateDnsManagementClient).private_zones
+    zone = PrivateZone(location='global', tags=tags)
+    return client.create_or_update(resource_group_name, private_zone_name, zone, if_none_match='*')
+
+
+def update_privatedns_zone(instance, tags=None):
+    if tags is not None:
+        instance.tags = tags
+    return instance
+
+
+def create_privatedns_link(cmd, resource_group_name, private_zone_name, virtual_network_link_name, virtual_network, registration_enabled, tags=None):
+    from azure.mgmt.privatedns import PrivateDnsManagementClient
+    from azure.mgmt.privatedns.models import VirtualNetworkLink
+    client = get_mgmt_service_client(cmd.cli_ctx, PrivateDnsManagementClient).virtual_network_links
+    link = VirtualNetworkLink(location='global', tags=tags)
+
+    if registration_enabled is not None:
+        link.registration_enabled = registration_enabled
+
+    if virtual_network is not None:
+        link.virtual_network = virtual_network
+
+    return client.create_or_update(resource_group_name, private_zone_name, virtual_network_link_name, link, if_none_match='*')
+
+
+def update_privatedns_link(instance, registration_enabled=None, tags=None):
+    if registration_enabled is not None:
+        instance.registration_enabled = registration_enabled
+
+    if tags is not None:
+        instance.tags = tags
+
+    return instance
+
+
+def create_privatedns_record_set(cmd, resource_group_name, private_zone_name, relative_record_set_name, record_type, metadata=None, ttl=3600):
+    from azure.mgmt.privatedns import PrivateDnsManagementClient
+    from azure.mgmt.privatedns.models import RecordSet
+    client = get_mgmt_service_client(cmd.cli_ctx, PrivateDnsManagementClient).record_sets
+    record_set = RecordSet(ttl=ttl, metadata=metadata)
+    return client.create_or_update(resource_group_name, private_zone_name, record_type, relative_record_set_name, record_set, if_none_match='*')
+
+
+def list_privatedns_record_set(cmd, resource_group_name, private_zone_name, record_type=None):
+    from azure.mgmt.privatedns import PrivateDnsManagementClient
+    client = get_mgmt_service_client(cmd.cli_ctx, PrivateDnsManagementClient).record_sets
+    if record_type is not None:
+        return client.list_by_type(resource_group_name, private_zone_name, record_type)
+
+    return client.list(resource_group_name, private_zone_name)
+
+
+def _privatedns_type_to_property_name(key):
+    type_dict = {
+        'a': 'a_records',
+        'aaaa': 'aaaa_records',
+        'cname': 'cname_record',
+        'mx': 'mx_records',
+        'ptr': 'ptr_records',
+        'soa': 'soa_record',
+        'srv': 'srv_records',
+        'txt': 'txt_records',
+    }
+    return type_dict[key.lower()]
+
+
+def _privatedns_add_record(record_set, record, record_type, is_list=False):
```

Comment:
`_privatedns_add_record` uses `_privatedns_type_to_property_name `, whereas `_add_record` uses `_type_to_property_name` and also, if in the future, the Dns team makes changes to these functions, it will affect our service as well, which we don't want.